AWSutils.sh

```diff
@@ -376,13 +378,51 @@ function InstallSSMagent {
    fi
 }
 
+# Install AWS CFN Bootstrap
+function InstallCfnBootstrap {
+   if [[ -z ${CFNBOOTSTRAP:-} ]]
+   then
+      err_exit "AWS CFN Bootstrap not requested for install. Skipping..." NONE
+   elif [[ ${CFNBOOTSTRAP} == *.tar.gz ]]
+   then
+      local TMPDIR
+      TMPDIR=$(chroot "${CHROOTMNT}" mktemp -d)
+
+      err_exit "Fetching ${CFNBOOTSTRAP}..." NONE
+      curl -sL "${CFNBOOTSTRAP}" -o "${CHROOTMNT}${TMPDIR}/aws-cfn-bootstrap.tar.gz"
+
+      err_exit "Installing AWS CFN Bootstrap..." NONE
+      chroot "${CHROOTMNT}" python3 -m pip install "${TMPDIR}/aws-cfn-bootstrap.tar.gz" || \
+        err_exit "Failed installing AWS CFN Bootstrap"
+
+      err_exit "Setting up directory structure for cfn-hup service..." NONE
+      chroot "${CHROOTMNT}" install -Ddm 000755 /opt/aws/apitools/cfn-init/init/redhat/ /opt/aws/bin || \
+         err_exit "Failing setting up cfn-hup directories"
+
+      err_exit "Creating symlink for cfn-hup executable..." NONE
+      chroot "${CHROOTMNT}" ln -sf /usr/local/bin/cfn-hup /opt/aws/bin/cfn-hup || \
```

Comment:
You _could_ use the `alternatives` system here, instead. Then, someone familiar with that subsystem could use the `alternatives` command's list-option to centrally-view any links that have been pre-created for them

Reply:
I'm kinda doing this the other way around... `pip install` creates the executable at `/usr/local/bin/cfn-hup`, so it's already the default. But the service init script expects the executable at `/opt/aws/bin/cfn-hup`. So I'm just creating the symlink so I don't have to modify the init script...

Reply:
Ok, updated to use alternatives to create this symlink, and manage the initscript